c8ylp/cli/core.py

```diff
@@ -633,12 +634,20 @@ def start_proxy(
     background = None
 
     try:
-        tcp_server = TCPProxyServer(
-            opts.port,
-            WebsocketClient(**client_opts),
-            opts.tcp_size,
-            opts.tcp_timeout,
-        )
+        if opts.socket_path is not None:
+            tcp_server = UnixStreamProxyServer(
```

Comment:
We should change the variable name from "tcp_server" to something more generic, like "server" since we now support different types of servers

Reply:
Ok, I did a few more commits to integrate your wishes,

* did the linting, formatting
* added a very basic unit test regarding the cli, didn't find any "real" test for the server except in the integration tests
* I customized the usage example on the cli if invoked with the socket-path option and added a small example to the readme